src/D2L.Security.OAuth2/Keys/Default/JsonWebKeySet.cs

```diff
@@ -20,10 +20,22 @@ internal sealed class JsonWebKeySet {
 
 				List<object> keyObjects = data["keys"];
 
-				var builder = ImmutableArray.CreateBuilder<JsonWebKey>();
+				var builder = ImmutableArray.CreateBuilder<JsonWebKey>(
+					initialCapacity: keyObjects.Count
+				);
+
 				foreach( object keyObject in keyObjects ) {
 					string keyJson = JsonConvert.SerializeObject( keyObject );
-					JsonWebKey key = JsonWebKey.FromJson( keyJson );
+
+					if( !JsonWebKey.TryParseJsonWebKey( keyJson, out var key, out var error, out var exception, out var useEncKey ) ) {
+						if( useEncKey ) {

```

Comment:
This filtering could apply more generally but for now I just want to do the minimal thing to get the integrations working. I think the API needs a redesign overall.